### PR TITLE
add 5-patch-article-votes-by-id

### PR DESCRIPTION
**Pull request by AlistairHopson**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -65,10 +65,86 @@ describe("GET /api/articles/:article_id", () => {
   test("requests for invalid article IDs are rejected", () => {
     return request(app)
       .get("/api/articles/999999")
-      .expect(404)
+      .expect(400)
```
Comment (douglashellowell) [inline __tests__/app.test.js:79]: GET /api/articles/99999999 wouldn't be a bad request - it is valid - it just so happens that particular id doesn't exist

File: __tests__/app.test.js [hunk 2]
```diff
@@ -65,10 +65,86 @@ describe("GET /api/articles/:article_id", () => {
...
+describe("PATCH /api/articles/:article_id", () => {
+  test("article votes can be increased", () => {
+    const addedVotes = { inc_votes: 55 };
+    return request(app)
+      .patch("/api/articles/9")
+      .send(addedVotes)
+      .expect(200)
+      .then(({ body }) => {
+        expect(Object.keys(body).length).toBe(1);
+        expect.objectContaining({
```
Comment (douglashellowell) [inline __tests__/app.test.js]: expect _what_ is an `objectContaining`?

you're missing some syntax here

make sure you're using the `expect` function and some sort of _matcher_ (i.e: `toEqual` etc.)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -65,10 +65,86 @@ describe("GET /api/articles/:article_id", () => {
...
+          title: "They're not exactly dogs, are they?",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "Well? Think about it.",
+          created_at: 1591438200000,
+          votes: -55,
+        });
+      });
+  });
+  test("article votes unchanged if inc_votes is 0", () => {
```
Comment (douglashellowell) [inline __tests__/app.test.js:125]: i'm a bit confused about this test

it's targeting the same article as the previous test with the same `inc_votes` but it's expecting a different result!

You are getting a false positive from your `objectContaining` mistake (see previous comment)

Needs a refactor

**Review by douglashellowell — CHANGES_REQUESTED** [review]
Need some fixes in your tests please!

There are other errors that might occur in a PATCH request

Think about how you'd want your server to respond if someone sent an incorrect body with their request (what if inc_votes wasn't a number etc.)

I've noticed an issue with a test from a previous PR

```js
 expect(body.topics.length > 0);
```
what are you expecting that to resolve to... true? false? 
make sure you are using jest matchers in your assertions!

**Comment by AlistairHopson** [comment]
Thanks for the feedback - I've amended these

**Review by Angaca — APPROVED** [review]
Ready to merge!

**Pull request merged**